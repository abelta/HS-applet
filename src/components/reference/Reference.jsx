import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Reference = ({ className, id, name, onClick }) => (
  <button
    className={classNames('row', 'btn-flat', 'waves-effect', 'waves-dark', 'white', className)}
    onClick={onClick}
    style={{ border: 'none', margin: 0, width: '100%' }}
  >
    <span className="col s2" style={{ fontWeight: 'bold' }}>
      {id}
    </span>
    <span className="col s10">{name}</span>
  </button>
);

Reference.defaultProps = {
  className: '',
  onClick: null,
};

Reference.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

export default Reference;
