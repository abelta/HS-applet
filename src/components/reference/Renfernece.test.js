import React from 'react';
import renderer from 'react-test-renderer';
import Reference from './Reference';

describe('Reference', () => {
  const tree = renderer.create(<Reference id="id" name="name" />).toJSON();

  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
