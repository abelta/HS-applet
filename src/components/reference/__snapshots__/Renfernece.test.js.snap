// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reference renders correctly 1`] = `
<button
  className="row btn-flat waves-effect waves-dark white"
  onClick={null}
  style={
    Object {
      "border": "none",
      "margin": 0,
      "width": "100%",
    }
  }
>
  <span
    className="col s2"
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    id
  </span>
  <span
    className="col s10"
  >
    name
  </span>
</button>
`;
